editoast/src/views/stdcm_search_environment.rs

```diff
@@ -169,13 +193,75 @@ async fn retrieve_latest(
     }
 }
 
+#[derive(IntoParams, Deserialize)]
+#[allow(unused)]
+struct StdcmSearchEnvIdParam {
+    /// An existing stdcm search environment ID
+    env_id: i64,
+}
+
+#[utoipa::path(
+    delete, path = "",
+    tag = "stdcm_search_environment",
+    params(StdcmSearchEnvIdParam),
+    responses(
+        (status = 204, description = "The stdcm search environment was deleted successfully"),
+        (status = 404, body = InternalError, description = "The requested stdcm search environment was not found"),
+    )
+)]
+async fn delete(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Path(StdcmSearchEnvIdParam { env_id }): Path<StdcmSearchEnvIdParam>,
+) -> Result<impl IntoResponse> {
+    let authorized = auth
+        .check_roles([Role::Admin].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+    StdcmSearchEnvironment::delete_static_or_fail(conn, env_id, || StdcmSearchEnvError::NotFound {
+        env_id,
+    })
+    .await?;
+
+    Ok(axum::http::StatusCode::NO_CONTENT)
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "stdcm_search_environment",
+    responses(
+        (status = 200, body = Vec<StdcmSearchEnvironment>, description = "The list of all existing stdcm search environments"),
+    )
+)]
+async fn list(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+) -> Result<Json<Vec<StdcmSearchEnvironment>>> {
+    let authorized = auth
+        .check_roles([Role::Admin].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+    let mut conn = db_pool.get().await?;
+    let listed_envs = StdcmSearchEnvironment::list(&mut conn, SelectionSettings::new()).await?;
```

Comment:
```suggestion
    let listed_envs = StdcmSearchEnvironment::list_paginated(&mut conn, SelectionSettings::new()).await?;
```

our listing endpoints are usually paginated, checkout the `PaginationQueryParams` struct.

Reply:
Hmm, I'm not sure paginating would be best here. I will if it's a requirement for api coherence (or another reason), but our usage here will require requesting all stdcm env (thus going through all pages if we paginate) - and the table is likely to be very small, so requests shouldn't be too large. 

Since you said usually I assume it's not a hard requirement, but if it is or you still prefer we paginate, just say so ^^

Reply:
I think we should systematically control the size of both requests and responses for api sanity. Especially those which can dump a whole table. We often paginate responses, but we never control the input - and yet we have a lot of batching endpoints. So in that sense we're not really consistent about sizing. 

I do agree with you about the number of items though, and since both the `POST` and `/list` only works for admins, I'm fine with dropping the pagination in this case.

---

However

> thus going through all pages if we paginate

Don't editoast customers have utils to go through all pages of a paginated response easily? :o Our pagination stats keys are the same for every paginated endpoint, an `async function unpaginate<T>(paginated_repsonse)` would surely be useful if it doesn't exist imo. Wdyt?

Reply:
> Don't editoast customers have utils to go through all pages of a paginated response easily? :o

Hmm, I had a look around the front to check, and it seems to me that for most list or search endpoints we simply assume there will be less than a 1000 items of anything we request (infras, projects, scenarios, rolling stocks...), and never request a second page. 

There's a few (timetables, macro nodes and nge ops) where we do iterate through all pages, but we implement it each time.

Having an unified way to do this could be nice I suppose. Since we need to make multiple requests, it would have to take a redux ```ApiEndpointQuery```as an argument though, not a ```paginated_response```, wouldn't it? Unless I'm misunderstanding your idea.

End to end tests have their own (simpler) way of talking to the api though, and we would simply need to provide an url as argument, so implementing a generic listApiRequest should be very easy. I guess I'll do it, to future proof things a bit.

Reply:
Thanks for the detailed answer! :)

I was aware that old components just expects all items to be returned, but I'm surprised that something as usual as pagination management is implemented manually each time on more recent UI.

> it would have to take a redux ApiEndpointQueryas an argument though

Probably, not sure what that is :p

Reply:
> not sure what that is

An ugly generated type from redux with way too many generic arguments that I'm glad I don't have to deal with in this pr =p